persist/fs/retriever.go

```diff
@@ -285,6 +308,23 @@ func (r *blockRetriever) Stream(
 	startTime time.Time,
 	onRetrieve block.OnRetrieveBlock,
 ) (xio.SegmentReader, error) {
+	// This should never happen
+	if len(r.seekerMgrs) < 1 {
+		return nil, errNoSeekerMgrs
+	}
+	// It doesn't matter which seekerManager we use (they're all identical, we
+	// just have multiple for concurrency reasons), so we use the first one
+	// because it's guaranteed to be there (concurrency cannot be < 1)
+	seeker, err := r.seekerMgrs[0].Seeker(shard, startTime)
+	if err != nil {
+		return nil, err
+	}
+	// If the ID is not in the seeker's bloom filter, then it's definitely not on
+	// disk and we can return immediately
+	if !seeker.ConcurrentIDBloomFilter().Test(id.Data().Get()) {
+		return nil, errSeekIDNotFound
```

Comment:
I'm not sure if we treat an error here as if it's a not found error.  I think traditionally we call back the `req.onRetrieved` with an empty segment (which is what it looks like by looking at `fetchBatch` call).

Reply:
I think you're right. Unfortunately the existing behavior for this method is not documented with a test, so I ended up writing one but it looks like I didn't understand the existing behavior properly. I'll try and write a better test that captures the existing codes intentions

Reply:
OK I think I got this all figured out, will update it soon. Out of curiosity, how come this didn't manifest in production? Is it because we use the index to lookup IDs first, so we almost always know an ID exists before we query for it?

Reply:
Gonna add a new integration test for this because we don't have one for querying an ID that does not exist

Reply:
@robskillington I tried to add an integration test for this, and I don't think I can because it relies too heavily on the ReadEncoded path which you're heavily modifying in your P.R. Could you add an integration test to your P.R that tests reading an ID from disk that doesn't exist and make sure it returns an empty array and not an error? Could also modify an existing integration test, I'll handle the unit test on my side